app/services/tunnel.py

```
@@ -1,13 +1,32 @@
+from subprocess import Popen, PIPE
 from typing import Optional
+import time
+
+import requests
 
 
 def start_tunnel(port: int) -> Popen:
     """Inicia o túnel e retorna o processo."""
-    # Implementação simplificada; em produção escolheria provider.
-    return Popen(["cloudflared", "tunnel", "--url", f"http://localhost:{port}"])
+    proc = Popen(
```

Ruff (0.12.2)

`subprocess` call: check for execution of untrusted input

(S603)